module/zfs/zvol.c

```diff
@@ -369,6 +369,41 @@ zvol_set_volsize(const char *name, uint64_t volsize)
 	return (SET_ERROR(error));
 }
 
+/*
+ * Update volthreading.
+ */
+int
+zvol_set_volthreading(const char *name, uint64_t value)
+{
+	zvol_state_t *zv = zvol_find_by_name(name, RW_NONE);
+	if (zv) {
+		zv->zv_threading = value;
+		mutex_exit(&zv->zv_state_lock);
+		return (0);
+	}
+	return (ENOENT);
```

Comment:
I don't remember the zvol code enough to recall in what race cases can this error happen, but I see a difference between error reporter here and not reported in zvol_prop_set_ro().  Is it intentional?

Aside of that, I think would the code be more readable if we invert the logic, returning immediately in case of zv == NULL and doing mutex_exit() at the end otherwise, more symmetric with zvol_find_by_name() that takes the lock.

Reply:
`zvol_prop_set_ro` is shared by both the filesystem and zvol type whereas volthreading property is only zvol type.
Updated as per your second suggestion.